service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateControlledResourceFlight.java

```diff
@@ -243,6 +250,7 @@ public CreateControlledResourceFlight(FlightMap inputParameters, Object beanBag)
 
   private void addNotebookSteps(
       UserWithPetSa userAndPet,
```

Comment:
It feels like we're passing in user informatino in three pieces, and it's not obvious which one is authoritative (i.e. is the user email supposed to be in the `userAndPet` or the `userRequest`. I guess just maybe comment on that for now, and think about a higher-level `UserBundle` object or something that knows the user, pet, and proxy.

Reply:
I think that is another piece of work. It would be useful to have a consistent collection of the parts. Perhaps we could extend the AuthenticatedUserRequest rather than make another structure to pass around. Just an idea.